compiler/rustc_driver/src/lib.rs

```diff
@@ -1040,28 +1040,28 @@ pub fn handle_options(args: &[String]) -> Option<getopts::Matches> {
         let unstable_enabled = nightly_options::is_unstable_enabled(&matches);
         let nightly_build = nightly_options::match_is_nightly_build(&matches);
         usage(matches.opt_present("verbose"), unstable_enabled, nightly_build);
-        return None;
+        return Ok(None);
     }
 
     // Handle the special case of -Wall.
     let wall = matches.opt_strs("W");
     if wall.iter().any(|x| *x == "all") {
         print_wall_help();
-        return None;
+        return Err(ErrorReported);
```

Comment:
You can use `rustc_errors::FatalError.raise();` instead. This is what `early_error()` uses too.

Reply:
Nice! That works. I've updated the PR.